src/IO/LasLoader/lasloader.cpp

```diff
@@ -0,0 +1,242 @@
+    vertices.reserve( nb_data );
+
+    // checking for colors
+    bool colors = false;
+    if ( data_format == 2 || data_format == 3 || ( minor >= 3 && data_format == 5 ) ) {
+        auto handle_color = geometry->getGeometry().vertexAttribs().addAttrib<Ra::Core::Vector4>(
+            Ra::Core::Geometry::getAttribName( Ra::Core::Geometry::MeshAttrib::VERTEX_COLOR ) );
+        colors = true;
+        LOG( logINFO ) << "\"red\", \"green\", \"blue\".";
+    }
+
+    // checking for GPS Time
+    bool gps_time = false;
+    if ( ( data_format == 1 ) || ( data_format == 3 ) ||
+         ( ( minor > 2 ) && ( data_format == 4 ) ) || ( ( minor > 2 ) && ( data_format == 5 ) ) ||
+         ( ( minor == 4 ) && ( data_format == 6 ) ) ) {
+        auto handle_time = geometry->getGeometry().vertexAttribs().addAttrib<Scalar>( "gps_time" );
+        gps_time         = true;
+        LOG( logINFO ) << "\"GPS time\".";
+    }
+
+    for ( unsigned int i = 0; i < nb_data; ++i ) {
+        // positioning at point data location in file
+        int pos = offset + i * data_len;
+        stream.seekg( 0 );
+        stream.seekg( pos );
+
+        // reading point data
+        stream.read( point.data(), data_len );
+
+        // extracting initial x y z coordinates
+        int ix;
+        int iy;
+        int iz;
+
+        ix = *(int*)( point.data() );
+        iy = *(int*)( point.data() + 4 );
+        iz = *(int*)( point.data() + 8 );
+
+        // computing actual coordinates
+        double x = (double)ix * scale_x + offset_x;
+        double y = (double)iy * scale_y + offset_y;
+        double z = (double)iz * scale_z + offset_z;
+
+        // storing coordinates in vertices object
+        vertices.emplace_back( Scalar( x ), Scalar( y ), Scalar( z ) );
+
+        // computing colors if found
+        if ( colors ) {
+            unsigned short red, green, blue;
+
+            if ( data_format == 5 || data_format == 3 ) {
+                red   = *(unsigned short*)( point.data() + 28 );
+                green = *(unsigned short*)( point.data() + 30 );
+                blue  = *(unsigned short*)( point.data() + 32 );
+            }
+            else {
+                red   = *(unsigned short*)( point.data() + 20 );
+                green = *(unsigned short*)( point.data() + 22 );
+                blue  = *(unsigned short*)( point.data() + 24 );
+            }
+
+            auto handle_colors =
+                geometry->getGeometry().vertexAttribs().findAttrib<Ra::Core::Vector4>(
+                    Ra::Core::Geometry::getAttribName(
+                        Ra::Core::Geometry::MeshAttrib::VERTEX_COLOR ) );
+            geometry->getGeometry()
+                .vertexAttribs()
+                .getDataWithLock( handle_colors )
+                .emplace_back( Scalar( red ), Scalar( green ), Scalar( blue ), 1_ra );
+            geometry->getGeometry().vertexAttribs().unlock( handle_colors );
```

Comment:
It will be more efficient to look for the attrib only once.
Note that you can do
```
            auto& attr =  geometry->getGeometry(). getAttrib( handle_colors );
```

I think that I now understand why you need your temporary arrays, my bad.
So:
 - use temporary std::vector for colors (and other attributes)
 - after reading, copy them using `geometry->getGeometry(). getAttrib( handle_colors ).setData()`.

Reply:
updated